### ISSUE TITLE
Push notifications for iOS Swift SDK

### ISSUE THREAD

**Issue by niklasribbon** [aws-amplify/aws-sdk-ios#4375]
**State your question**
I see you now list push notifications as one of the features of this SDK. And I can do amplify add notifications to add the p12 APNS cert etc. But I can find zero documentation on how to use this?

I suppose I can just upload the device token to my models, then use this form a Lambda function to call AWSPinpoint to send a push?

But is there some way of doing this with your SDKs?

**Comment by ameter** [comment]
@niklasribbon Yes, you can use the AWSPinpoint client in the SDK for integration with Pinpoint push notifications.  See the [push notifications documentation](https://docs.amplify.aws/sdk/push-notifications/getting-started/q/platform/ios/) here for details.

**Comment by niklasribbon** [comment]
Hi thanks for prompt reply. I'm still struggling with how to be able to pass the device token to pinpoint. This line of code form the docs.

`/ Register the device token with Pinpoint as the endpoint for this user
    pinpoint?.notificationManager
        .interceptDidRegisterForRemoteNotifications(withDeviceToken: deviceToken)
}`

I tried to use the escape hatch:
`do {
    // Retrieve the reference to AWSPinpointAnalyticsPlugin
    let plugin = try Amplify.Analytics.getPlugin(for: "awsPinpointAnalyticsPlugin")
    guard let analyticsPlugin = plugin as? AWSPinpointAnalyticsPlugin else {
        return
    }
    
    // Retrieve the reference to PinpointClientProtocol
    let pinpointClient = analyticsPlugin.getEscapeHatch()

    // Make requests using pinpointClient...
    // ...
} catch {
    print("Get escape hatch failed with error - \(error)")
}`

But then using this pinpointClient to call notificationManager.interceptDidRegisterForRemoteNotifications(withDeviceToken: deviceToken) doesn't work?


If I try to push to the device token I have from pinpoint console nothing happens. So I assume I have to give the device token to point point first?

**Comment by ameter** [comment]
Ah, ok, so you're using Amplify as opposed to using the aws-sdk-ios directly.  You are correct that you need to update the endpoint address (i.e. the APNs token) with Pinpoint before you can push to the device.  Which version of Amplify are you using?

**Comment by niklasribbon** [comment]
@ameter yeah the old libraries don't even seem to be available as SwiftPackages, and I assume it'll be conflicted to use both. I'm on the latest v2.0.0.  

I'm actually now able to send a push directly from a Lambda function, so doesn't seem like I need to give the device token to pinpoint, unless I want to run campaign and segments. It does however seem like I manually have to go in and give IAM permission for my amplify added function to be allowed to do sendMessages on Pinpoint.

**Comment by ameter** [comment]
>I see that in v2.0.1 it has "Make AWSPluginsCore public (https://github.com/aws-amplify/aws-sdk-ios/issues/2472)"... So maybe I can access this now...

No, as of v2.0.0, Amplify no longer uses [aws-sdk-ios](https://github.com/aws-amplify/aws-sdk-ios).  It now leverages the new [aws-sdk-swift](https://github.com/awslabs/aws-sdk-swift).

AWS Pinpoint Push Notifications support is part of [aws-sdk-ios](https://github.com/aws-amplify/aws-sdk-ios)'s AWSPinpoint, which in Amplify v1.x is provided via Analytics' escape hatch.

As Amplify v2.x has moved away from aws-sdk-ios, Pinpoint Push Notifications support is not currently available.
However, we are currently looking into it.
